src/lib/decoders.ts

```diff
@@ -0,0 +1,118 @@
+import { Result } from '@ethersproject/abi';
+import BigNumber from 'bignumber.js';
+import { BytesLike, defaultAbiCoder } from 'ethers/lib/utils';
+import { parseInt } from 'lodash';
+import { BN_0 } from '../bignumber-constants';
+import { NULL_ADDRESS } from '../constants';
+import { MultiResult } from './multi-wrapper';
+
+export function generalDecoder<T>(
+  result: MultiResult<BytesLike>,
+  types: string[],
+  defaultValue: T,
+  parser?: (v: Result) => T,
+): T {
+  if (!result.success || result.returnData === '0x') {
+    return defaultValue;
+  }
+  const decoded = defaultAbiCoder.decode(types, result.returnData);
+  return parser ? parser(decoded) : decoded[0];
+}
+
+export const uintDecode = (result: MultiResult<string>): bigint => {
+  if (!result.success) {
+    return 0n;
+  }
+  return BigInt(defaultAbiCoder.decode(['uint'], result.returnData)[0]);
+};
+
+export const uint256ArrayDecode = (result: MultiResult<string>): bigint => {
+  if (!result.success) {
+    return 0n;
+  }
+  return defaultAbiCoder
+    .decode(['uint256[]'], result.returnData)[0]
+    .map((r: BigNumber) => BigInt(r.toString()));
+};
+
+export const uin256DecodeToBigNumber = (
+  result: MultiResult<string>,
+): BigNumber => {
+  if (!result.success) {
+    return BN_0;
+  }
+  return new BigNumber(
+    defaultAbiCoder.decode(['uint256'], result.returnData)[0],
+  );
+};
+
+export const uin256DecodeToNumber = (result: MultiResult<string>): number => {
+  if (!result.success) {
+    return 0;
+  }
+  return parseInt(
+    defaultAbiCoder.decode(['uint256'], result.returnData)[0],
+    10,
+  );
+};
+
+export const uin256DecodeToFloat = (result: MultiResult<string>): number => {
+  if (!result.success) {
+    return 0;
+  }
+  return parseFloat(defaultAbiCoder.decode(['uint256'], result.returnData)[0]);
+};
+
+export const uin128DecodeToFloat = (result: MultiResult<string>): number => {
+  if (!result.success) {
+    return 0;
+  }
+  return parseFloat(defaultAbiCoder.decode(['uint128'], result.returnData)[0]);
+};
```

Comment:
Don't understand why we are using `parseFloat` or having float methods at all (floats and ints are both just `number`).  The ABI coder will always return integers that will parse with `parseInt`.

Reply:
I don't know. Better to check in the @Louis-Amas POC PR, where it is used. I just copied it here for consistency

Reply:
I added this methods when migrating curve I think we were using parseFloat there but no idea why

Reply:
Let's remove that if it's useless